ghcide/src/Development/IDE/Plugin/TypeLenses.hs

```diff
@@ -187,19 +212,21 @@ suggestLocalSignature isQuickFix mEnv mTmr mBindings Diagnostic{_message, _range
     , startOfLine <- Position (_line _start) startCharacter
     , beforeLine <- Range startOfLine startOfLine
     , title <- if isQuickFix then "add signature: " <> signature else signature
-    , action <- TextEdit beforeLine $ signature <> "\n" <> T.replicate (fromIntegral startCharacter) " " =
+    , range' <- fromMaybe beforeLine (flip toCurrentRange beforeLine =<< mmp)
```

Comment:
```suggestion
    , Just range' <- flip toCurrentRange beforeLine =<< mmp
```
?

Reply:
Or rather, the logic seems slightly off. You want to return `beforeLine` if `mmp` is `Nothing`, but fail the guard if `toCurrentRange` is `Nothing`, I think.

Reply:
Yes, I want to return `beforeLine` if `mmp` is Nothing or `toCurrentRange` returns Nothing, the guard seems won't fail.🤔

Reply:
I think we shouldn't return if `toCurrentRange` returns `Nothing`. That means we have a mapping but we can't get it to work. Rather I think the guard should just fail! Same as here: https://github.com/haskell/haskell-language-server/pull/3570#discussion_r1174427561

So it's something like:
```
Just range' <- case mmp of { Nothing -> Just beforeLine; Just mp -> toCurrentRange mp beforeLine; }
```